code/game/objects/structures/window.dm

```diff
@@ -641,6 +670,19 @@
 	explosion_block = 1
 	glass_type = /obj/item/stack/sheet/rglass
 
+/obj/structure/window/full/reinforced/polarized
+	name = "electrochromic window"
+	desc = "Adjusts its tint with voltage. Might take a few good hits to shatter it."
+	var/id
+
+/obj/structure/window/full/reinforced/polarized/proc/toggle()
```

Comment:
Since this is a duplicate of `/obj/structure/window/reinforced/polarized/proc/toggle()`, could this maybe be placed at the base window level? Something like `/obj/structure/window/proc/toggle_polarization()`.